pkg/catalogv2/oci/client.go

```
@@ -0,0 +1,316 @@
+	}
+
+	// Fetch the manifest blob of the oci artifact
+	manifestblob, err := content.FetchAll(ctx, memoryStore, manifest)
+	if err != nil {
+		return "", false, fmt.Errorf("unable to fetch the manifest blob of %s:%w", ociURL, err)
+	}
+	var manifestJSON ocispecv1.Manifest
+	err = json.Unmarshal(manifestblob, &manifestJSON)
+	if err != nil {
+		return "", false, fmt.Errorf("unable to unmarshal manifest blob of %s:%w", ociURL, err)
+	}
+
+	// Create a temp file to store the helm chart tar
+	tempFile, err := os.CreateTemp("", "helm-")
+	if err != nil {
+		return "", false, fmt.Errorf("unable to create temp file for storing the oci artifact")
+	}
+
+	// Check if the oci artifact is of type helm config ?
+	if manifestJSON.ArtifactType == helmregistry.ConfigMediaType || manifestJSON.Config.MediaType == helmregistry.ConfigMediaType {
+		// find the layer of helm chart type and fetch it
+		for _, layer := range manifestJSON.Layers {
+			if layer.MediaType == helmregistry.ChartLayerMediaType {
+				chartTar, err := content.FetchAll(ctx, memoryStore, layer)
+				if err != nil {
+					return "", false, fmt.Errorf("unable to fetch chart blob of %s:%w", ociURL, err)
+				}
+
+				err = os.WriteFile(tempFile.Name(), chartTar, 0o600)
+				if err != nil {
+					return "", false, fmt.Errorf("unable to write chart %s into file %s:%w", ociURL, tempFile.Name(), err)
+				}
+
+				return tempFile.Name(), true, nil
+			}
+		}
+	}
+
+	return tempFile.Name(), false, nil
+}
+
+// getAuthClient creates an oras auth client that can be used
+// in creating an oras registry client or oras repository client.
+func (o *Client) getAuthClient() (http.Client, error) {
+	config := &tls.Config{
+		InsecureSkipVerify: o.insecure,
+	}
+	if len(o.caBundle) > 0 {
+		cert, err := x509.ParseCertificate(o.caBundle)
+		if err != nil {
+			return http.Client{}, err
+		}
+		pool, err := x509.SystemCertPool()
+		if err != nil {
+			pool = x509.NewCertPool()
```

I am not sure we want to ignore this error. I can understand why we would do this before [go1.18](https://go-review.googlesource.com/c/go/+/353589) for Windows, otherwise I would rather return the error. Am I missing something here?

---

Added a debug statement to log the error message instead of ignoring it.